src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,121 @@
+package duke;
+
+import duke.commands.AddCommand;
+import duke.commands.ClearCommand;
+import duke.commands.Command;
+import duke.commands.DeleteCommand;
+import duke.commands.ExitCommand;
+import duke.commands.FindCommand;
+import duke.commands.ListCommand;
+import duke.commands.MarkCommand;
+import duke.commands.UnmarkCommand;
+import duke.exceptions.CommandDetailException;
+import duke.exceptions.CommandNotRecognizedException;
+import duke.exceptions.TimeParsingException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+
+/**
+ * Represents a parser for interpreting user input commands.
+ * The parser is responsible for translating string input into executable command objects.
+ */
+public class Parser {
+
+    public Parser() {
+    }
+
+    /**
+     * Parses the user input string and translates it into an executable command object.
+     *
+     * @param input The string input from the user.
+     * @return A Command object representing the user's intent.
+     * @throws TimeParsingException          If there's an error in parsing the time for tasks.
+     * @throws CommandNotRecognizedException If the entered command is not recognized.
+     * @throws CommandDetailException        If there's an error in the command details.
+     */
+    public Command parse(String input)
+            throws TimeParsingException, CommandNotRecognizedException, CommandDetailException {
+        String trimmedInput = input.trim();
+        String[] splitInput = trimmedInput.split(" ", 2);
+        String command = splitInput[0].toLowerCase();
+        try {
+            switch (command) {
+            case "bye": {
+                return new ExitCommand();
+            }
+            case "list": {
+                return new ListCommand();
+            }
+            case "mark": {
+                int markIndex = Integer.parseInt(splitInput[1]) - 1;
+                return new MarkCommand(markIndex);
+            }
+            case "delete": {
+                int deleteIndex = Integer.parseInt(splitInput[1]) - 1;
+                return new DeleteCommand(deleteIndex);
+            }
+            case "unmark": {
+                int markIndex = Integer.parseInt(splitInput[1]) - 1;
+                return new UnmarkCommand(markIndex);
+            }
+            case "find": {
+                String keyword = splitInput[1];
+                return new FindCommand(keyword);
+            }
+            case "clear": {
+                return new ClearCommand();
+            }
+            default: {
+                Task task = parseTask(trimmedInput);
+                if (task != null) {
+                    return new AddCommand(task);
+                } else {
+                    throw new CommandNotRecognizedException("I'm sorry, but I don't know what " + input + " means :-(");
```

Comment:
This line might be a little too long too. Perhaps you could separate the string concatenation on different lines?

```java 
{
  throw new CommandNotRecognizedException("I'm sorry, but I don't know what " 
      + input 
          + " means :-(");
}
```